Ability to turn on / off the logger from the GUI.

ktf opened the pull request:
Ability to turn on / off the logger from the GUI.

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/AliceO2Group/AliceO2/pull/12529).
* #12533
* __->__ #12529 (5 commits)

alibuild commented:
Error while checking build/O2/fullCI for a9f53d8aee1c66fa8af02efbfd633efab5081e5d at 2024-01-16 10:46:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2-full-system-test-latest/log
command /sw/slc8_x86-64/O2/12529-slc8_x86-64-local1/prodtests/full-system-test/dpl-workflow.sh had nonzero exit code 128
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data HMP/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data FLP/DISTSUBTIMEFRAME/0 was not created for timeslice 0 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data MID/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data CTP/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data TOF/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data TPC/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data ITS/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data MFT/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data MCH/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data PHS/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data TRD/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data CPV/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data EMC/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data FDD/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data FT0/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data FV0/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data ZDC/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data FLP/DISTSUBTIMEFRAME/52443 was not created for timeslice 1 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data HMP/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data FLP/DISTSUBTIMEFRAME/0 was not created for timeslice 1 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data MID/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data CTP/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data TOF/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data TPC/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data ITS/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data MFT/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data MCH/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data PHS/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data TRD/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data CPV/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data EMC/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data FDD/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data FT0/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data FV0/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[48362:raw-file-reader]: [09:45:53][ERROR] Expected Lifetime::Timeframe data ZDC/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[48387:BadMapCalibSpec]: [09:46:28][ERROR] Insufficient statistics: 3 entries in lowE histo, do nothing
[48465:ctf-writer]: [09:46:30][ERROR] Some Lifetime::Timeframe data got dropped starting at 1
[ERROR] Workflow crashed - PID 48446 (tpc-calibrator-gainmap-tracks) did not exit correctly however it's not clear why. Exit code forced to 128.
[ERROR]  - Device tpc-calibrator-gainmap-tracks: pid 48446 (exit 128)
[ERROR] SEVERE: Device tpc-calibrator-gainmap-tracks (48446) returned with 128
[0 more errors; see full log]
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/12529/a9f53d8aee1c66fa8af02efbfd633efab5081e5d/build_O2_fullCI/pretty.html).

alibuild commented:
Error while checking build/O2/fullCI for 12f7e25fe2f434c30605ae94b52ce7e89e7300de at 2024-01-16 14:13:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2-full-system-test-latest/log
command /sw/slc8_x86-64/O2/12529-slc8_x86-64-local1/prodtests/full-system-test/dpl-workflow.sh had nonzero exit code 128
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data HMP/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data FLP/DISTSUBTIMEFRAME/0 was not created for timeslice 0 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data MID/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data CTP/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data TOF/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data TPC/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data ITS/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data MFT/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data MCH/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data PHS/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data TRD/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data CPV/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data EMC/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data FDD/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data FT0/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data FV0/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data ZDC/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data FLP/DISTSUBTIMEFRAME/52443 was not created for timeslice 1 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data HMP/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data FLP/DISTSUBTIMEFRAME/0 was not created for timeslice 1 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data MID/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data CTP/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data TOF/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data TPC/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data ITS/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data MFT/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data MCH/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data PHS/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data TRD/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data CPV/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data EMC/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data FDD/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data FT0/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data FV0/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[71184:raw-file-reader]: [13:12:30][ERROR] Expected Lifetime::Timeframe data ZDC/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[71209:BadMapCalibSpec]: [13:12:46][ERROR] Insufficient statistics: 4 entries in lowE histo, do nothing
[71367:ctf-writer]: [13:12:50][ERROR] Some Lifetime::Timeframe data got dropped starting at 1
[ERROR] Workflow crashed - PID 71323 (tpc-calibrator-gainmap-tracks) did not exit correctly however it's not clear why. Exit code forced to 128.
[ERROR]  - Device tpc-calibrator-gainmap-tracks: pid 71323 (exit 128)
[ERROR] SEVERE: Device tpc-calibrator-gainmap-tracks (71323) returned with 128
[0 more errors; see full log]
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/12529/12f7e25fe2f434c30605ae94b52ce7e89e7300de/build_O2_fullCI/pretty.html).

ktf committed:
DPL use Signposts to debug consumeWhenAll

ktf committed:
DPL: drop unused o2_signpost_id_make_with_pointer method

Probably old remaining from a previous implementation.

ktf committed:
DPL: rework signposts support

The logger based Signposts are now behaving the same on macOS and linux when
running normally.

The difference between the two implementations will now be that if Instruments
enables some OSLog namespace, we stop printing the messages and use the os_signpost
API instead.

ktf committed:
DPL: add proper tracing for DataProcessingDevice socket callbacks

ktf committed:
Ability to turn on / off the logger from the GUI.

alibuild commented:
Error while checking build/O2/fullCI for ab530c0085516f3640ae4b2ca7ec349b8f86d2b9 at 2024-01-16 21:19:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2-full-system-test-latest/log
task timeout reached .. killing all processes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data HMP/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data FLP/DISTSUBTIMEFRAME/0 was not created for timeslice 0 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data MID/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data CTP/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data TOF/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data TPC/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data ITS/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data MFT/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data MCH/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data PHS/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data TRD/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data CPV/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data EMC/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data FDD/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data FT0/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data FV0/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data ZDC/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data FLP/DISTSUBTIMEFRAME/52443 was not created for timeslice 1 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data HMP/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data FLP/DISTSUBTIMEFRAME/0 was not created for timeslice 1 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data MID/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data CTP/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data TOF/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data TPC/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data ITS/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data MFT/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data MCH/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data PHS/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data TRD/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data CPV/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data EMC/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data FDD/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data FT0/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data FV0/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[204233:raw-file-reader]: [19:49:13][ERROR] Expected Lifetime::Timeframe data ZDC/RAWDATA was not created for timeslice 1 and might result in dropped timeframes


## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
[0 more errors; see full log]
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/12529/ab530c0085516f3640ae4b2ca7ec349b8f86d2b9/build_O2_fullCI/pretty.html).
